functionapp.tf

```
@@ -14,7 +14,7 @@ resource "azurerm_app_service_plan" "asp" {
 
 resource "azurerm_key_vault" "kv" {
   location                  = var.location
-  name                      = "acme${lower(data.azurerm_subscription.subscriptionid.display_name)}"
+  name                      = "acme${replace(local.name, "sharedservices", "sds")}"
```

would be nice to keep dashes (-) in key vault names as it makes it easier to read

```suggestion
  name                      = "acme-${local.shortened_subscription_name}"
```